resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ResteasyClientBuilderImpl.java

```diff
@@ -420,7 +413,8 @@ public ResteasyClient build()
       if (serviceLoaderIterator.hasNext()) {
          config.register(new ClientConfigProviderFilter(serviceLoaderIterator.next()), Priorities.AUTHENTICATION);
       }
-      return createResteasyClient(engine, executor, cleanupExecutor, scheduledExecutorService, config);
+      final ContextualExecutorService executor = getExecutorService();
+      return createResteasyClient(engine, executor, !executor.isManaged(), ContextualExecutors.wrapOrLookup(scheduledExecutorService), config);

```

Comment:
When create a `ResteasyClient` without setting a `ScheduledExecutorService`, the `ContextualExecutors.wrapOrLookup() `will always find the` ScheduledExecutorService` from a JNDI context after check the delegate is null. For the most ResteasyClient which is created and running in real client side instead of in container, this JNDI look up can be a lot of cost.  Should we optimize this to skip this lookup when it's not in container environment ?

Reply:
@jimma I couldn't find a good way to determine if the environment was in a Jakarta EE container or not. What I did was created a static map to hold the lookup names and a boolean of whether or not the lookup failed or succeeded. In environments that don't contain the Jakarta EE Concurrency utilities the lookup will only happen once.

Reply:
@jamezp  OK. This already improved one part thing. Let's look at if there is something we can do more to improve this in the futurte.

Reply:
Sounds good to me @jimma. We can definitely improve this. It could be interesting to see if we can figure out a way to indicate whether we are running in a container or not.